### PR TITLE
Fixed broken dependencies for apache-beam.

### PR DESCRIPTION
**Pull request by potiuk**
Apache beam is now also included in our extras.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Review by mik-laj — APPROVED** [review]

**Comment by potiuk** [comment]
Got it even more stable and likely even works with 20.3.1 (not even master)

File: setup.cfg [hunk 1]
```diff
@@ -116,10 +117,12 @@ install_requires =
     markdown>=2.5.2, <4.0
     markupsafe>=1.1.1, <2.0
     marshmallow-oneofschema>=2.0.1
+    mock<3.0.0  # required to keep apache-beam happy
     pandas>=0.17.1, <2.0
     pendulum~=2.0
     pep562~=1.0;python_version<"3.7"
     psutil>=4.2.0, <6.0.0
+    pyarrow<0.18.0
```
Comment (ashb) [inline setup.cfg]: This is quite old, I'm not sure we _want_ this globally.

Comment (ashb) [reply setup.cfg]: (Pyarrow is now at 2.0.0 for instance.)

**Comment by ashb** [comment]
My concern here is this pins a lot of things to old versions globally, to support a sub-provider (not even a provider) that is only used by a tiny subset of our users.

**Comment by turbaszek** [comment]
> tiny subset of our users.

Depends how we count, Google Dataflow operators are ones of most commonly used from Google provider.

@potiuk have you try to upgrade beam? I think latest 2.X version has higher reuirements

File: setup.cfg (hunk 1 above)
Comment (potiuk) [reply setup.cfg]: Yeah. I will experiment with moving those out of install_requires. I want to see if they will work for our tests first

**Comment by potiuk** [comment]
> > tiny subset of our users.
> 
> Depends how we count, Google Dataflow operators are ones of most commonly used from Google provider.
> 
> @potiuk have you try to upgrade beam? I think latest 2.X version has higher requirements

Higher versions of beam have other conflicts, unfortunately unsolvable with other providers (I think requests). We have to limit it to 2.23

**Comment by potiuk** [comment]
FYI: The master PIP also does not work with those constraints (even if they are non-conflicting) everything installs fine in pip 20.2.44, but the new PIP has problem with it:

```
Collecting apache-beam[gcp]<=2.23.0
  Downloading apache_beam-2.23.0-cp36-cp36m-manylinux2010_x86_64.whl (8.3 MB)
     |████████████████████████████████| 8.3 MB 22.8 MB/s
INFO: pip is looking at multiple versions of analytics-python to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of amqp to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of alembic to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of sphinxcontrib-spelling to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of sphinx-autoapi to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of rich to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of apscheduler to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of python-telegram-bot to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of pylint to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of pinotdb to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of mypy to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of json-merge-patch to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of flask-swagger to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of elasticsearch-dbapi to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of colorlog to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of sphinx-airflow-theme to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of apache-airflow to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of apache-airflow[devel-ci] to determine which version is compatible with other requirements. This could take a while.
ERROR: Could not find a version that satisfies the requirement cachetools<4,>=3.1.0; extra == "gcp" (from apache-beam[gcp])
ERROR: No matching distribution found for cachetools<4,>=3.1.0; extra == "gcp"
```
It does not download all the many versions of packages, but it fails on cachetools this time. 

So either those are imaginary dependencies by mistakenly using some extras, or 2.20.4 did not detect those conflicts.

File: setup.cfg (hunk 1 above)
Comment (potiuk) [reply setup.cfg]: I only need to constraint deps in "install_requires" if they are already there transitively. I "eagerly" added them here to check if it works but likely httplib can be moved out, let's see pyarrow as well

Comment (potiuk) [reply setup.cfg]: Yep. All three are NOT transitive dependencies of airflow without extras, so It looks like we can move them where they belong.  Working on it.

Comment (potiuk) [reply setup.cfg]: The only one that will have to stay there is dill, but I think this is not a problem

**Comment by potiuk** [comment]
Confirmed. It works when I move httplib/pyarrow/mock limitation  to beam extr @ashb !

**Comment by potiuk** [comment]
So our tests and constraints will be running old versions of those libs, but this is not at all a problem. Constraints will also contain those, but this is not a problem as well, maybe we need to add few words about "users are free to upgrade the libraries as they see fit after installation" .

**Comment by potiuk** [comment]
Wohoo! Seems I've done that. Waiting for https://github.com/potiuk/airflow/actions/runs/410667474 but it looks like we have apache-beam included and no more exclusions @mik-laj and @turbaszek !

**Comment by potiuk** [comment]
Need to wait for snowflake connector

**Pull request closed without merging**